File: GamerHub_Backend/Entities/Producto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace GamerHub_Backend.Entities
{
    public class Producto
    {
        public int Id { get; set; }
        [Required]
        public string NombreProducto { get; set; } = null!;
        public byte[]? Imagen { get; set; }
        [Required]
        public decimal Precio { get; set; }
        [Required]
        public string Caracteristicas { get; set; } = null!;
        public string? Descripcion { get; set; }
        [Required]
        public int Stock { get; set; }
        [Required]
        public int CategoriaId { get; set; }
        public Categoria? Categoria { get; set; } = null!;
        [JsonIgnore]
        public List<DetallesCompra>? DetallesCompras { get; set; }
        [JsonIgnore]
        public List<Devolucion>? Devoluciones { get; set; }
        [JsonIgnore]
        public List<Comentario>? Comentarios { get; set; }
    }
}

File: GamerHub_Store/src/app/interface/Producto.ts
import { Categoria } from "./Categoria";
import { Comentario } from "./Comentario";
import { DetallesCompra } from "./DetallesCompra";
import { Devolucion } from "./Devolucion";

export interface Producto {
    id?: number;
    nombreProducto: string;
    imagen?: Uint8Array;
    precio: number;
    caracteristicas: string;
    descripcion: string;
    stock: number;
    categoriaId: number;
    categoria?: Categoria;
    detallesCompras?: DetallesCompra[];
    devoluciones?: Devolucion[];
    comentarios?: Comentario[];
}
